src/sentry/api/serializers/rest_framework/notification_action.py

```diff
@@ -77,6 +77,19 @@ def validate_trigger_type(self, trigger_type: str) -> int:
             )
         return trigger_type_value
 
+    def validate(self, data: NotificationActionInputData) -> NotificationActionInputData:
+        registration = NotificationAction.get_registration(
+            trigger_type=data["trigger_type"],
+            service_type=data["service_type"],
+            target_type=data["target_type"],
+        )
+        if not registration:
+            raise serializers.ValidationError(
+                "Combination of trigger_type, service_type and target_type has not been registered."
+            )
+        registration.validate_action(data=data)
+        return data
+
```

Comment:
This will ensure that new notification actions are registered before saving them to the DB. It will also give registrations the ability to validate their registered trigger/service/target combo before it gets saved to the database if they implement `validate_action`